packages/mask/src/extension/popups/pages/Personas/components/ProfileList/index.tsx

```diff
@@ -88,26 +119,113 @@ export const ProfileList = memo(() => {
         [],
     )
 
-    const [, onDisconnect] = useAsyncFn(async (identifier: ProfileIdentifier) =>
-        Services.Identity.detachProfile(identifier),
+    const onDisconnect = useCallback(
+        (identifier: ProfileIdentifier, is_valid?: boolean, platform?: NextIDPlatform, identity?: string) => {
+            if (is_valid) {
+                setUnbind({
+                    identifier,
+                    platform,
+                    identity,
+                })
+                return
+            }
+            Services.Identity.detachProfile(identifier)
+        },
+        [],
     )
 
+    const { value: mergedProfiles, retry: refreshProfileList } = useAsyncRetry(async () => {
+        if (!currentPersona) return []
+        const publicHexKey = await Services.Helper.queryPersonaHexPublicKey(currentPersona.identifier)
```

Comment:
The publicHex has on Persona type. Maybe we can add a property on PersonaInformation.